RUFAS/biophysical/animal/herd_manager.py

```diff
@@ -863,116 +863,120 @@ def initialize_pens(
+        pen receives any remaining animals to guarantee full allocation.
 
-        An alternative approach would be to allocate animals to the pens with the higher allocation limits first.
-        This would use up the animal count more quickly, so the later the allocation, the fewer animals are left
-        to allocate. Depending on the dynamics between the given numbers, some pens may end up with a very low density.
+        Overstocking is permitted when necessary and is handled fairly based on capacity-derived
+        allocation limits. Logging ensures that overstocked pens are tracked for review.
 
         Parameters
         ----------
         num_animals : int
-            The total number of animals to allocate. Must be a non-negative integer and not be greater than the
-            total number of spaces.
-        max_spaces_in_pens : List[int]
-            A list of integers representing the number of maximum spaces in each pen. Each integer must be positive.
+            The total number of animals to allocate. Must be a non-negative integer.
+        max_spaces_in_pens : list[int]
+            A list of integers representing the maximum number of animals each pen can accommodate
+            without overstocking. Each integer must be positive.
+        simulation_day : int
+            The current simulation day, used for warning logs when overstocking occurs.
 
         Returns
         -------
-        List[int]
-            A list of integers representing the allocation of animals in each pen. Each integer will be less than or
-            equal to `math.ceil(overall_density * max_spaces_in_pens[i])]`.
+        list[int]
+            A list of integers representing the number of animals allocated to each pen.
+            Each value may exceed the pen's capacity if overstocking occurs.
 
         Raises
         ------
-        ValueError
-            If the number of animals is greater than the total number of spaces.
+        AssertionError
+            If the total number of allocated animals does not match the number of animals provided.
 
         Examples
         --------
-        >>> HerdManager._plan_animal_allocation(num_animals=90, max_spaces_in_pens=[50, 30, 20])
-        [45, 27, 18]
+        >>> _plan_animal_allocation(num_animals=90, max_spaces_in_pens=[30, 30, 30], simulation_day=1)
+        [30, 30, 30]
 
-        >>> HerdManager._plan_animal_allocation(num_animals=70, max_spaces_in_pens=[50, 30, 20])
-        [35, 21, 14]
-
-        >>> HerdManager._plan_animal_allocation(num_animals=47, max_spaces_in_pens=[50, 30, 20])
-        [22, 15, 10]
+        >>> _plan_animal_allocation(num_animals=95, max_spaces_in_pens=[30, 30, 30], simulation_day=1)
+        [32, 32, 31]
 
+        >>> _plan_animal_allocation(num_animals=47, max_spaces_in_pens=[20, 15, 10], simulation_day=1)
+        [22, 15, 10]  # Overstocked due to animal count exceeding total capacity
```

Comment:
```suggestion
        >>> _plan_animal_allocation(num_animals=47, max_spaces_in_pens=[20, 15, 10], simulation_day=1)
        [20, 16, 11]  # Overstocked due to animal count exceeding total capacity
```
This example would actually output [20, 16,11]